hs_core/hydroshare/utils.py

```diff
@@ -601,16 +603,25 @@ def resource_modified(resource, by_user=None, overwrite_bag=True):
                 user = User.objects.get(username=by_user)
             except User.DoesNotExist:
                 user = None
+                error_message = f"Resource cannot be modified by nonexistent user '{by_user}'"
+
     if user:
-        resource.last_changed_by = user
+        # user can only mark modified if they are an owner of the resource
+        # this prevents the last_changed_by getting set to an admin user that edits a resource
+        user_privilege = resource.raccess.get_effective_user_privilege(user, ignore_superuser=True)
+        if user_privilege > PrivilegeCodes.CHANGE:
```

Comment:
If the user who has edit permission, adds files or updates metadata for the resource, we are not going update `resource.last_changed_by` and `resource.updated`?

Reply:
thanks Pabitra, good catch! You're right that this should've been `user_privilege < PrivilegeCodes.CHANGE`.

Reply:
Actually, I forgot that our PrivilegeCodes are "backwards" (higher means less privilege) so I think this was right before. Reverted it back.

Reply:
I think your original implementation (`user_privilege > PrivilegeCodes.CHANGE`) was correct. Sorry my confusion.

Reply:
No worries at all, thanks for making me look at it closer!